pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -751,6 +755,24 @@ func (aws *AWSCloud) InstanceID(name string) (string, error) {
 	}
 }
 
+// InstanceType returns the type of the specified instance.
+func (aws *AWSCloud) InstanceType(name string) (string, error) {
+	awsInstance, err := aws.getSelfAWSInstance()
+	if err != nil {
+		return "", err
+	}
+
+	if awsInstance.nodeName == name {
+		return awsInstance.instanceType, nil
+	} else {
+		inst, err := aws.getInstanceByNodeName(name)
+		if err != nil {
+			return "", err
+		}
+		return orEmpty(inst.InstanceType), nil
```

Comment:
FYI, you can use aws.StringValue instead of orEmpty (I wasn't aware of this, or it wasn't originally available).  No need to change though, just for future reference